### ISSUE TITLE
Issue with config module and pip install

### ISSUE THREAD

**Issue by aperezhortal** [pySTEPS/pysteps#21]
The default configuration parameters are now hardcoded in the **config** module. This means that when a user install a new version of pysteps using pip, the user-defined configuration will be overide.

To keep the user-defined configuration unchanged under library updates, an approach similar to the one used in Matplolib can be used to mantain the configuration files.
Matplotlib uses matplotlibrc configuration files to customize all kinds of propertieslike figure size and dpi, line width, color and style, axes, axis and grid properties, text and font properties and so on.
This gives the ability to the user to customize the default behavior of the library, with the advantage that this behavior is maintained when the user updates the library.

As a proof of concept, I adapt this idea to pysteps in [the following fork](https://github.com/aperezhortal/pysteps/tree/pysteps_new_config):

The datasources in the **config module** are now defined in a **pystepsrc** [JSON file ](https://en.wikipedia.org/wiki/JSON)

The JSON files resemble the definition of a dictionary in python, and allows to define many data types.

When pysteps is imported, it looks for pystepsrc file in four locations, in the following order:
    - `$PWD/pystepsrc`
    - `$PYSTEPSRC` if it is a file
    - `$PYSTEPSRC/pystepsrc`
    - `$HOME/.pysteps/pystepsrc` if `$HOME` is defined.
    - Lastly, it looks inside the library in `pysteps/pystepsrc` for a system-defined copy.

[This is the example default configuration file](https://github.com/aperezhortal/pysteps/blob/pysteps_new_config/pysteps/pystepsrc) that is included with the library.

Usage example:
```
from pysteps import rcparams
ds = rcparams["bom"]
```
The ds is an [AttrDict](https://pypi.org/project/attrdict/) object and it is compatible with the actual config module implementation to maintain backwards compatibility.

This functionality was implemented in the **pysteps/__init__.py** file.

Things that are still missing or maybe are nice to include
---------------------------------------------------------------------

* The global parameters defined in **pysteps_config.py** are not yet defined, but they can be easily implemented.
* If the user-defined file is loaded and some of the parameters where not defined, the default ones defined in the package are not used. The implementation should load first the package defaults, and the overide the ones that the user defines.

**Comment by dnerini** [comment]
I think this is a very useful development and in addition to what @aperezhortal already mentioned, there is also the advantage that the user will be able to modify the config files also after the pip install, while currently this is possible only before.

I've tested the code and it works very nicely. 

One concern I have is that it is not directly clear to the user which config file is being used and where this file comes from. This could be a bit confusing in case that multiple copies exist. But I don't know how this could be improved, other than printing a message with the path to the config file that it's imported.

We should also define how we recommended it to be used, as sometimes just giving too many options can be confusing. In the installation guide, we should include a step where the user is told how to create and customize a new config file (for example in his $HOME). 

Thanks for the nice piece of code, I hope that will be able to soon include it in pysteps!

**Comment by aperezhortal** [comment]
> One concern I have is that it is not directly clear to the user which config file is being used and where this file comes from. This could be a bit confusing in case that multiple copies exist. But I don't know how this could be improved, other than printing a message with the path to the config file that it's imported.

I pushed some modifications to my branch. 
I followed the Daniele's message suggestion, when pysteps is imported it prints a message showing the rc file used (commit 17480e265b72b2e58e46168044baef5b1fd03798).

> We should also define how we recommended it to be used, as sometimes just giving too many options can be confusing. In the installation guide, we should include a step where the user is told how to create and customize a new config file (for example in his $HOME).

As a draft, I updated the README.rst file with improved installation instructions and explaining the steps to setup the default parameters file on windows (tested), Linux (tested), and OS X (not tested). See commit 27f131821657c9e2f660757d99bc7fba62d22914.

**Comment by aperezhortal** [comment]
I worked the **pystepsrc** implementation a little bit further before I submitted a pull request: #22. The changes are in a new clean and up-to-date branch [here](https://github.com/aperezhortal/pysteps/tree/pysteps_json_config)).

Since many scripts and users may use the interface in the config module, I added the new parameters file support and adapted the config module  to read the parameters from the **pystepsrc** configuration file. The files inside the config module are no longer needed, only the **__init__.py** where the backward compatibility is implemented.
This means that all the examples works with out any modification. I just added a *FutureWarning* saying that the config module will be deprecated, in a future release. But not necessarily in the next one, making the transition to the rcparams interface smooth.

I also implemented a validator for the  **pystepsrc**, to ensure that has all the necessary fields with the correct data type.

Below the complete list of changes:
* All the parameters in config are now included in the pystepsrc file that uses the JSON format.
* The installation instruction where updated to include the correct setup of the pystepsrc file for each platform.
* The config module interface was maintained for backward compatibility, but it returns the parameters in the *pystepsrc** file.
* A JSON schema validator was implemented. If the **pystepsrc** has an error, an exception is raised during the import of the pysteps module. One more dependency was added for this (jsonschema).
* The requirements.txt file was added and the environment.yml was updated with the new dependencies.
* The README.rst , setup.py and MANIFEST.IN files where updated.
